maven-compat/src/main/java/org/apache/maven/artifact/repository/metadata/AbstractRepositoryMetadata.java

```diff
@@ -29,8 +29,8 @@
 import org.apache.maven.artifact.repository.ArtifactRepositoryPolicy;
 import org.apache.maven.artifact.repository.metadata.io.xpp3.MetadataXpp3Reader;
 import org.apache.maven.artifact.repository.metadata.io.xpp3.MetadataXpp3Writer;
-import org.codehaus.plexus.util.ReaderFactory;
-import org.codehaus.plexus.util.WriterFactory;
+import org.codehaus.plexus.util.xml.XmlStreamReader;
```

Comment:
org.apache.commons.io.input.XmlStreamReader is preferred here

Reply:
I don't want to introduce a new dependency for one class.  This one is in plexus-xml which won't go away any time soon.  We may be able to get rid of plexus-utils, but not plexus-xml which contains classes that are part of the maven 3 api.

Reply:
It;s not just about going away. I have more expectation that the version in apache commons will be maintained and correct than the one in plexus.

Reply:
Actually, I think the usage of those classes is just wrong and they already have caused issues with tycho.  So I'll modify this PR to simply use plain InputStream / OutputStream.

Reply:
best of all possible worlds. +1